src/Sparrow/Json/BlittableJsonReaderObject.cs

```diff
@@ -822,7 +822,10 @@ private object GetObjectUnlikely(BlittableJsonToken type, int position, Blittabl
                 case BlittableJsonToken.EmbeddedBlittable:
                     return ReadNestedObject(position);
                 case BlittableJsonToken.StartArray:
-                    return new BlittableJsonReaderArray(position, _parent ?? this, type)
+                    return new BlittableJsonReaderArray(position, _parent ?? this, type,
```

Comment:
I do not think we want such a behavior? All child objects are working on the same memory fragment and it was always an assumption that clone needs to be done when you want to 'detach' child object from parent.